R/xml_nodes_to_lints.R

```diff
@@ -0,0 +1,89 @@
+#' Convert an XML node or nodeset into a Lint
+#'
+#' Convenience function for converting nodes matched by XPath-based
+#'   linter logic into a [Lint()] object to return.
+#'
+#' @inheritParams lint-s3
+#' @param xml An `xml_node` object (to generate one `Lint`) or an
+#'   `xml_nodeset` object (to generate several `Lint`s), e.g. as returned by
+#'   [xml2::xml_find_all()] or [xml2::xml_find_first()].
+#' @param source_expression A source expression object, e.g. as
+#'   returned typically by [lint()], or more generally
+#'   by [get_source_expressions()].
+#' @param lint_message The message to be included as the `message`
+#'   to the `Lint` object. If `lint_message` is a `function`,
+#'   this function is first applied to `xml` (so it should be a
+#'   function taking an `xml_node` as input and must produce a
+#'   length-1 character as output). If `lint_message` is a character vector the same length as `xml`,
```

Comment:
do we want to add a test to enforce the equal-lengths part more strictly?

Reply:
Nonequal lengths will generate a recycling warning by `.mapply()`, right?

Reply:
right... tidyverse usually hates recycling though and tends to error instead

Reply:
Sounds like a good candidate for a sanity-check follow up (in addition to the `start <= column <= nchar(line)` check discussed below.